R/distances.R

```diff
@@ -419,41 +430,104 @@ distancesMatrix <- function(t.layer, starters = NULL, targets = starters,
     length(suppressWarnings(packageDescription("terra"))))
   missing.packages <- sapply(aux, function(x) x == 1)
   if (any(missing.packages)) {
-    stop(paste0("This function requires packages '", paste(c("raster", "gdistance", "sp", "terra")[missing.packages], collapse = "', '"),
-      "' to operate. Please install ", ifelse(sum(missing.packages) > 1, "them", "it"), " before proceeding.\n"), call. = FALSE)
+    stop(paste0("This function requires packages '", 
+                paste(c("raster", "gdistance", "sp", "terra")[missing.packages], 
+                      collapse = "', '"),
+      "' to operate. Please install ", 
+      ifelse(sum(missing.packages) > 1, "them", "it"), 
+      " before proceeding.\n"), call. = FALSE)
   }
 
   if (!inherits(t.layer, "TransitionLayer"))
-    stop("Could not recognise 't.layer' as a TransitionLayer object. Make sure to compile it using the function transitionLayer.\n", call. = FALSE)
+    stop("Could not recognise 't.layer' as a TransitionLayer object. Make",
+         " sure to compile it using the function transitionLayer.\n", 
+         call. = FALSE)
 
   if (!is.null(id.col) && length(id.col) > 1)
     stop("Please provide only one column name in 'id.col'", call. = FALSE)
   if (!is.null(id.col) && is.numeric(id.col))
-    stop("Please refer to the column name in 'id.col', rather than the column index.\n", call. = FALSE)
+    stop("Please refer to the column name in 'id.col', rather than the column",
+         " index.\n", call. = FALSE)
 
   if (actel) {
     message("M: Creating actel-compatible distances matrix."); flush.console()
-    if (!is.null(starters) | !is.null(targets))
-      warning("starters' or 'targets' were set but will be ignored because 'actel' is set to TRUE. Set 'actel' to FALSE to use the 'starters' and 'targets' arguments.", call. = FALSE, immediate. = TRUE)
-    starters <- targets <- loadSpatial()
-    if (!is.null(id.col))
-      warning("id.col' was set but will be ignored because 'actel' is set to TRUE. Set 'actel' to FALSE to use the 'id.col' argument.", call. = FALSE, immediate. = TRUE)
+    
+    if(!is.null(starters) | !is.null(targets)) {
+      # distancesMatrix() crashes if actel = TRUE and starters/targets were 
+      # supplied but do not contain a column called "Standard.name"
+      if(!is.null(starters) & !any(grepl(c("Standard.name"), 
+                                         colnames(starters)))) {
+        event("'starters' is missing column called 'Standard.name'",
+              type = "stop")
```

Comment:
```suggestion
        stop("'starters' is missing column called 'Standard.name'",
             call. = FALSE)
```

distancesMatrix operates outside the actel analyses, so it doesn't need to use event() - at least for now. Best to keep it consistent until we decide we need to change everything to events.